File: Smodr/src/main/java/com/kevintcoughlin/smodr/models/Channel.java
package com.kevintcoughlin.smodr.models;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.ElementList;
import org.simpleframework.xml.Path;
import org.simpleframework.xml.Root;
import org.simpleframework.xml.Text;

import java.util.List;

@Root(name = "channel", strict = false)
public class Channel {
    Channel() {

    }

    public Channel(String title, String link, String imageUriString) {
        this.title = title;
        this.link = link;
        this.image = new Image(imageUriString);
    }

    @Element
    public String title;

    @Element(required = false)
    public String link;

    @Path("image")
    public Image image;

    @ElementList(inline = true, entry = "item")
    public List<Item> item;
}

File: Smodr/src/main/java/com/kevintcoughlin/smodr/models/Channel.kt
package com.kevintcoughlin.smodr.models

/**
 * Represents a podcast channel containing metadata and episodes.
 *
 * @property title The title of the podcast channel
 * @property link The URL link to the podcast channel
 * @property items The list of episodes/items in this channel
 */
data class Channel(
    val title: String? = null,
    val link: String? = null,
    val items: List<Item> = emptyList()
)